docs/plugin-protocol/tfplugin5.9.proto

```diff
@@ -677,7 +677,14 @@ message MoveResourceState {
         // The private state of the resource being moved.
         bytes source_private = 6;
 
-        ResourceIdentityData source_identity = 7;
+        // The raw identity of the resource being moved. Only the json field is
+        // populated, as there should be no legacy providers using the flatmap
+        // format that support newly introduced RPCs.
+        RawState source_identity = 7;
```

Comment:
Should we be consistent with `UpgradeResourceIdentity` here and just use `bytes`?

Reply:
Maybe `UpgradeResourceIdentity` should be consistent with the existing calls and use `RawState`?

Reply:
Either/or 😆 

I actually dropped a comment on the RFC recently about potential "future" data in identity and whether that would potentially be part of these types of RPCs: https://docs.google.com/document/d/1jvNfineZgXaKFP5djLkMxXKf9s6Xq-FzBV3p2jHGbiI/edit?disco=AAABe3pkd1Y

If that seems like something we might do, perhaps a new `RawIdentity` would fit as well

Reply:
Good catch! I'll update it the other way around to be consistent. 

The idea there was that if we ever added a new base encoding that wasn't `json`, it could be added to that message type.

Reply:
`RawState` is just meant to represent raw encoded data from the state file, and not imply any meaning of that data. The encoded identity is just a chunk of the resource's state, with the same encoding as the state in general, hence `RawState`. In reality `RawState` was required at the time of the 0.11-0.12 transition because there were multiple state encodings, and it's also been maintained since because however unlikely there could a different encoding in the future. We could decide that all raw state encoded data _must_ be json, but I think that's outside the scope here, and we should use the existing patterns in place until then ;)

Reply:
Fair enough, sounds good to me 👍🏻